pkg/dns/dns.go

```diff
@@ -0,0 +1,45 @@
+package dns
+
+import (
+	"errors"
+
+	"github.com/openshift/installer/pkg/types/aws"
+	"github.com/openshift/installer/pkg/types/azure"
+	"github.com/openshift/installer/pkg/types/libvirt"
+	"github.com/openshift/installer/pkg/types/none"
+	"github.com/openshift/installer/pkg/types/openstack"
+)
+
+//ConfigProvider is an interface that provides means to fetch the DNS settings
+type ConfigProvider interface {
+	GetBaseDomain() (string, error)
+	GetPublicZone(name string) string
```

Comment:
Even this is going to be used by assets that use installconfig asset as input to perform base_domain -> public_zone. so it makes sense to keep them in `pkg/asset/installconfig`

Reply:
moved